### PR TITLE
CI: Add support for GHC 8.10, stack and live-server testing

### PR DESCRIPTION
**Pull request by donatello**

### CODE REVIEW

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -31,24 +32,71 @@ jobs:
...
-    - name: Install dependencies
+    - name: Before install (Linux)
+      if: matrix.os == 'ubuntu-latest'
+      run: |
+        mkdir -p /tmp/minio /tmp/minio-config/certs
+        cp test/cert/* /tmp/minio-config/certs/
+        (cd /tmp/minio; wget https://dl.min.io/server/minio/release/linux-amd64/minio; chmod +x ./minio)
+        sudo cp /tmp/certs/public.crt /usr/local/share/ca-certificates/
+        sudo update-ca-certificates
+        MINIO_ACCESS_KEY=minio MINIO_SECRET_KEY=minio123 /tmp/minio/minio server --quiet --certs-dir /tmp/certs data 2>&1 > minio.log &
```
Comment (harshavardhana) [inline .github/workflows/haskell.yml:64]: enable multi-disk so that we can eventually test versioning as well @donatello ?

Comment (donatello) [reply .github/workflows/haskell.yml:64]: Sure, if I get this basic thing working, I'll get to that.

File: .github/workflows/haskell-cabal.yml [hunk 2]
```diff
@@ -13,18 +13,30 @@ jobs:
...
-          # Appears to be buggy to build in windows
+        os: [ubuntu-latest, macOS-latest]
+        experimental: [false]
+        include:
+          - ghc: 8.6
+            os: windows-latest
+            experimental: false
+          - ghc: 8.10
+            os: windows-latest
+            experimental: false
```
Comment (donatello) [inline .github/workflows/haskell-cabal.yml:31]: These are the same by default right? Don;t think we need these.

File: .github/workflows/haskell-stack.yml [hunk 3]
```diff
@@ -13,18 +13,29 @@ jobs:
...
+            os: windows-latest
+            experimental: false
+          - ghc: 8.10
+            os: windows-latest
+            experimental: false
+
+          # Appears to be buggy to build in windows with ghc 8.4 and 8.8
+          - ghc: 8.4
+            os: windows-latest
+            experimental: true
```
Comment (donatello) [inline .github/workflows/haskell-stack.yml:35]: In stack we are only building with 8.8, so we dont need these as well.

**Comment by donatello** [comment]
Windows builds that fail due to an unrelated error have been marked experimental, so CI can enforce that other build configurations must succeed. We will be merging this now, and making a release subsequently.

**Pull request merged**